File: Backend/Models/Dog.cs
namespace Backend.Models;

public class Dog
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public int Age { get; set; }
    public Guid BreedId { get; set; }
    public Guid OwnerId { get; set; }
    
    // Navigation property for the owner and breed
    public Owner Owner { get; set; } = null!;
    public Breed Breed { get; set; } = null!;
}

File: iOSDev/DoggoGO/Models/Dog.swift
import Foundation

struct Dog: Codable, Identifiable {
    let id: UUID
    let name: String
    let age: Int
    let breedId: UUID
    let ownerId: UUID?
    let breedName: String?
}

struct CreateDogRequest: Codable {
    let name: String
    let age: Int
    let breedId: UUID
}

struct UpdateDogRequest: Codable {
    let name: String?
    let age: Int?
    let breedId: UUID?
}